airflow/config_templates/default_airflow.cfg

```diff
@@ -640,18 +640,23 @@ logs_volume_host =
 # Git credentials and repository for DAGs mounted via Git (mutually exclusive with volume claim)
 git_repo =
 git_branch =
+git_subpath =
+# Use git_user and git_password for user authentication or git_ssh_key_secret_name and git_ssh_key_secret_key
+# for SSH authentication
 git_user =
 git_password =
-git_subpath =
 git_sync_root = /git
 git_sync_dest = repo
 # Mount point of the volume if git-sync is being used.
 # i.e. {AIRFLOW_HOME}/dags
 git_dags_folder_mount_point =
+# Kubernetes secret name and secret key where the SSH key is stored
+git_ssh_key_secret_name =
+git_ssh_key_secret_key =
```

Comment:
What is the key_secret_key? I can't work this out from having no context in to git-sync :)

Reply:
In a Kubernetes secret definition:
```
apiVersion: v1
kind: Secret
metadata:
  name: airflow-secrets
data:
  gitSshKey: <base64encodeddata>

git_ssh_key_secret_name -> airflow-secrets
git_ssh_key_secret_key    -> gitSshKey
```

Should I add this as a comment ?

Reply:
Final version could be:

```# Git-sync SSH authentication
# i.e. Kubernetes secrets:
# apiVersion: v1
# kind: Secret
# metadata:
#   name: airflow-secrets
# data:
#   gitSshKey: <base64_encoded_data>
#   knownHosts: <base64_encoded_data>
#
# git_ssh_key_secret_name = airflow-secrets
# git_ssh_key_secret_key  = gitSshKey
# git_ssh_known_hosts_secret_name = airflow-secrets
# git_ssh_known_hosts_secret_key = knownHosts
# Kubernetes secret name and secret key where the SSH key is stored
git_ssh_key_secret_name =
git_ssh_key_secret_key =
# Kubernetes secret name and secret key where the SSH known hosts file data is stored
git_ssh_known_hosts_secret_name =
git_ssh_known_hosts_secret_key =```

Reply:
Ah the "dict key" inside the secret. Does it actually need to be configurable, or could we just say "the secret must have look like "this"?

known_hosts could probably be a config map, not a secret (as it contains public keys, not private). Again - do we need the `*_key` suffix for known hosts either?

Reply:
I can enforce some specific key names, and point taken on known_hosts I'll change to a config map now

Reply:
I could reduce it to 1 config for each, by using the same syntax as in [kubernetes_secrets]

`secret-or-configmap-name=key-in-config`

Reply:
```